cves/CVE-2013-2135.yml

```diff
@@ -101,9 +114,11 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer:
-  code:
-  fix:
+  answer: |
+    Unit tests for the affected classes existed but did not test for the involved issues.
+    The fix commits included the addition of new unit tests to check for both issues involved in this vulnerability.
```

Comment:
Do you think unit tests are the appropriate fix here or are they a short term fix? Could modifying their design have been a stronger fix and a more mitigative approach?

Reply:
I believe the added unit tests were appropriate for fixing the double evaluation issue, as that fell under a simple coding mistake.

For the input validation component of this issue, I'm not familiar enough with Struts to say whether or not a design change would be better.